annif/backend/yake.py

```diff
@@ -0,0 +1,196 @@
+"""Annif backend using Yake keyword extraction"""
+# TODO Mention GPLv3 license also here?
+
+import yake
+import os.path
+import re
+from collections import defaultdict
+from rdflib.namespace import SKOS, RDF, OWL
+import rdflib
+import annif.util
+from . import backend
+from annif.suggestion import SubjectSuggestion, ListSuggestionResult
+from annif.exception import ConfigurationException
+
+
+class YakeBackend(backend.AnnifBackend):
+    """Yake based backend for Annif"""
+    name = "yake"
+    needs_subject_index = False
+
+    # defaults for uninitialized instances
+    _index = None
+    _graph = None
+    INDEX_FILE = 'yake-index'
+
+    DEFAULT_PARAMETERS = {
+        'max_ngram_size': 4,
+        'deduplication_threshold': 0.9,
+        'deduplication_algo': 'levs',
+        'window_size': 1,
+        'num_keywords': 100,
+        'features': None,
+        'default_label_types': ['pref', 'alt'],
+        'remove_specifiers': False
```

Comment:
I think `remove_qualifiers` would be a better name for the option, or perhaps `remove_parentheses`.